extension/observer/ecsobserver/docker_label.go

```diff
@@ -45,3 +50,70 @@ func (d *DockerLabelConfig) Init() error {
 	}
 	return nil
 }
+
+func (d *DockerLabelConfig) NewMatcher(options MatcherOptions) (Matcher, error) {
+	return &DockerLabelMatcher{
+		logger: options.Logger,
+		cfg:    *d,
+	}, nil
+}
+
+// DockerLabelMatcher implements Matcher interface.
+// It checks PortLabel from config and only matches if the label value is a valid number.
+type DockerLabelMatcher struct {
+	logger *zap.Logger
+	cfg    DockerLabelConfig
+}
+
+func (d *DockerLabelMatcher) Type() MatcherType {
+	return MatcherTypeDockerLabel
+}
+
+// MatchTargets first checks the port label to find the expected port value.
+// Then it checks if that port is specified in container definition.
+// It only returns match target when both conditions are met.
+func (d *DockerLabelMatcher) MatchTargets(t *Task, c *ecs.ContainerDefinition) ([]MatchedTarget, error) {
+	portLabel := d.cfg.PortLabel
+
+	// Only check port label
+	ps, ok := c.DockerLabels[portLabel]
+	if !ok {
+		return nil, errNotMatched
+	}
+
+	// Convert port
+	s := aws.StringValue(ps)
+	port, err := strconv.Atoi(s)
+	if err != nil {
+		return nil, fmt.Errorf("invalid port_label value, container=%s labelKey=%s labelValue=%s: %w",
+			aws.StringValue(c.Name), d.cfg.PortLabel, s, err)
+	}
+
+	// Checks if the task does have the container port
+	portExists := false
+	for _, portMapping := range c.PortMappings {
+		if aws.Int64Value(portMapping.ContainerPort) == int64(port) {
+			portExists = true
+			break
+		}
+	}
+	if !portExists {
+		return nil, errNotMatched
+	}
+
+	// Export only one target based on docker port label.
+	target := MatchedTarget{
+		Port: port,
+	}
+	if v, ok := c.DockerLabels[d.cfg.MetricsPathLabel]; ok {
```

Comment:
What if `metricPath` is not configured or invalid? We don't verify it util the scraper starts to collect the metrics?

Reply:
There is no good way to validate it as it can be any string. I think prometheus itself has the logic to retry/backoff when an invalid point is given to the scrape manger from service discovery manger.

We can validate port because it should be a numeric string.